src/ZendeskApi.Client/ZendeskClient.cs

```diff
@@ -1,53 +1,57 @@
 ﻿using System;
-using ZendeskApi.Client.Http;
+using Microsoft.Extensions.Logging;
 using ZendeskApi.Client.Resources;
-using ILogAdapter = ZendeskApi.Client.Logging.ILogAdapter;
-using ISerializer = ZendeskApi.Client.Serialization.ISerializer;
-
 
 namespace ZendeskApi.Client
 {
-    public class ZendeskClient : ClientBase, IZendeskClient
+    public class ZendeskClient : IZendeskClient
     {
-        public ITicketResource Tickets { get; private set; }
-        public ITicketCommentResource TicketComments { get; private set; }
-        public IRequestCommentResource RequestComments { get; private set; }
-        public IOrganizationResource Organizations { get; private set; }
-        public ISearchResource Search { get; private set; }
-        public IGroupResource Groups { get; private set; }
-        public IAssignableGroupResource AssignableGroups { get; private set; }
-        public IUserResource Users { get; private set; }
-        public IUserIdentityResource UserIdentities { get; private set; }
-        public IUploadResource Upload { get; private set; }
-        public ITicketFieldResource TicketFields { get; private set; }
-        public ITicketFormResource TicketForms { get; private set; }
-        public IOrganizationMembershipResource OrganizationMemberships { get; private set; }
-        public IRequestResource Request { get; private set; }
-        public ISatisfactionRatingResource SatisfactionRating { get; private set; }
-
-        public ZendeskClient(
-            Uri baseUri, 
-            ZendeskDefaultConfiguration configuration, 
-            ISerializer serializer = null, 
-            IHttpChannel httpChannel = null, 
-            ILogAdapter logger = null)
-            :base(baseUri, configuration, serializer, httpChannel, logger)
+        private readonly IZendeskApiClient _apiClient;
+        private readonly ILogger _logger;
+
+        public ZendeskClient(IZendeskApiClient apiClient, ILogger logger)
```

Comment:
Im wondering how i feel about a required(non-optional) logger.

On one hand its best practice but on the other enforces more complex setup of the client for lighter/exploratory usage of the client. I would like to be able to just get up and running without a logger if that was my choice as a consumer - what do you think?